docs/docs/reference/metaprogramming/inline.md

```diff
@@ -0,0 +1,226 @@
+---
+layout: doc-page
+title: Inline
+---
+
+## Inline (whitebox/blackbox)
+
+`inline` is a new [soft modifier](../soft-modifier.html) that guarantees that a
+definition will be inlined at the point of use. Example:
+
+```scala
+object Config {
+  inline val logging = false
+}
+
+object Logger {
+
+  private var indent = 0
+
+  inline def log[T](msg: => String)(op: => T): T =
+    if (Config.logging) {
+      println(s"${"  " * indent}start $msg")
+      indent += 1
+      val result = op
```

Comment:
Maybe use the safer `val result = try op finally indent -= 1` idiom here.

Reply:
I think it is sufficient for the exposition inlining